R/task-create.R

```diff
@@ -36,14 +39,18 @@
 ##' @export
 task_create_explicit <- function(expr, export = NULL, envir = .GlobalEnv,
                                  environment = "default", submit = NULL,
-                                 root = NULL) {
+                                 resources = NULL, root = NULL) {
   root <- hipercow_root(root)
+  if (is.null(resources)) {
+    resources <- hipercow_resources()
+  }
+  resources <- hipercow_resources_validate(resources, names(root$config), root)
```

Comment:
These 4 lines will be everywhere. Add a new helper:

```
as_hipercow_resources <- function(x) {
  if (is.null(resources)) {
    resources <- hipercow_resources()
  }
  hipercow_resources_validate(resources, names(root$config), root)
}
```

in resources.R; there's more good we can do with this later too

Reply:
Done - as_hipercow_resources(resources, root)